File: code/R/11_fig1_study_design.R
## Fig 1 — Study design schematic
## A4-landscape vector PDF, no external image dependencies.
## Device: macOS native quartz(type="pdf") for full Unicode + true sub/superscripts
## (no XQuartz / cairo dependency).
## Layout: 4 stages stacked top-to-bottom
##   Stage 1: 5 datasets (4 diseases, n = 856)
##   Stage 2: uniform pipeline (DESeq2/limma → meta → corr → mediation)
##   Stage 3: three analytical questions
##   Stage 4: principal finding (composition mediation ≈ 90 %)

suppressPackageStartupMessages({
  library(grid)
})

# ---- helper to draw a labeled box ----
# label may be a character string OR a plotmath/expression object
box_grob <- function(x, y, w, h, label, fill = "white", col = "black",
                     fontsize = 9, fontface = "plain", lwd = 1) {
  grobTree(
    rectGrob(x = x, y = y, width = w, height = h,
             gp = gpar(fill = fill, col = col, lwd = lwd)),
    textGrob(label, x = x, y = y,
             gp = gpar(fontsize = fontsize, fontface = fontface,
                       lineheight = 0.95))
  )
}

# ---- helper to draw an arrow between two points ----
arrow_grob <- function(x0, y0, x1, y1, lwd = 1) {
  segmentsGrob(x0 = x0, y0 = y0, x1 = x1, y1 = y1,
               gp = gpar(lwd = lwd),
               arrow = arrow(type = "closed", length = unit(0.10, "inches")))
}

out_pdf <- "results/figures/Fig1_study_design.pdf"
dir.create(dirname(out_pdf), recursive = TRUE, showWarnings = FALSE)

# macOS-native vector PDF — full Unicode, no XQuartz, embedded fonts
quartz(type = "pdf", file = out_pdf, width = 11.69, height = 8.27)

grid.newpage()
pushViewport(viewport(x = 0.5, y = 0.5, width = 0.96, height = 0.94))

# ----- Title -----
grid.text("Cross-disease re-analysis of the IDI2-AS1 / IL5 axis in IL5-driven allergic disease",
          x = 0.5, y = 0.97,
          gp = gpar(fontsize = 14, fontface = "bold"))
grid.text("Five public bulk RNA-seq cohorts, 856 samples, uniform pipeline",
          x = 0.5, y = 0.935,
          gp = gpar(fontsize = 10, fontface = "italic", col = "grey30"))

# ----- Stage 1 — 5 dataset boxes -----
grid.text("STAGE 1 \u2014 Datasets",
          x = 0.07, y = 0.88, hjust = 0,
          gp = gpar(fontsize = 10, fontface = "bold", col = "steelblue4"))

ds <- list(
  list("Asthma\nGSE152004\nn = 695\nNasal epithelium",     "#D6EAF8"),
  list("Atopic dermatitis\nGSE121212\nn = 65\nSkin biopsy", "#FADBD8"),
  list("EoE\nGSE246323\nn = 10\nEsophagus",                 "#FCF3CF"),
  list("EoE\nGSE58640\nn = 16\nEsophagus",                  "#FCF3CF"),
  list("CRSwNP\nGSE136825\nn = 70\nNasal polyp",            "#D5F5E3")
)
xs <- seq(0.10, 0.90, length.out = 5)
for (i in seq_along(ds)) {
  grid.draw(box_grob(xs[i], 0.80, 0.16, 0.10,
                     label = ds[[i]][[1]],
                     fill = ds[[i]][[2]], fontsize = 8))
}

# ----- Single horizontal aggregator bar Stage 1 -> Stage 2 -----
# Avoids the messy 5-arrows-converging look of the previous version.
grid.lines(x = c(xs[1], xs[5]), y = c(0.745, 0.745),
           gp = gpar(lwd = 0.8, col = "grey40"))
for (i in seq_along(ds)) {
  grid.lines(x = c(xs[i], xs[i]), y = c(0.75, 0.745),
             gp = gpar(lwd = 0.8, col = "grey40"))
}
grid.draw(arrow_grob(0.5, 0.745, 0.5, 0.71, lwd = 0.9))

# ----- Stage 2 — pipeline -----
grid.text("STAGE 2 \u2014 Uniform pipeline",
          x = 0.07, y = 0.69, hjust = 0,
          gp = gpar(fontsize = 10, fontface = "bold", col = "steelblue4"))

# Use bquote/expression for the IDI2-AS1 ↔ IL5 cell so the arrow renders
# as a proper Unicode glyph (already U+2194 here; quartz embeds it).
pipe <- list(
  "DESeq2 + apeglm\n(raw counts)\n  /  limma-trend\n(FPKM)",
  "Random-effects\nmeta-analysis\n(metafor REML)",
  "Sample-level\nSpearman\nIDI2-AS1 \u2194 IL5",
  "Cell-type-adjusted\npartial correlation\n(eosinophil + Th2\nsignatures)",
  "Causal mediation\n(asthma cohort,\n1,000 bootstraps)"
)
xp <- seq(0.10, 0.90, length.out = 5)
for (i in seq_along(pipe)) {
  grid.draw(box_grob(xp[i], 0.61, 0.17, 0.10,
                     label = pipe[[i]], fill = "#EAF2F8", fontsize = 7.5))
  if (i < length(pipe)) {
    grid.draw(arrow_grob(xp[i] + 0.085, 0.61, xp[i + 1] - 0.085, 0.61, lwd = 0.8))
  }
}

# ----- Stage 3 — three questions -----
grid.text("STAGE 3 \u2014 Three in vivo questions",
          x = 0.07, y = 0.50, hjust = 0,
          gp = gpar(fontsize = 10, fontface = "bold", col = "steelblue4"))

q <- c(
  "Q1.\nIs IDI2-AS1 itself\ndifferentially\nexpressed in disease\ntissue?",
  "Q2.\nDo IDI2-AS1 and IL5\ncovary at the\nsample level\nin the predicted\n(negative) direction?",
  "Q3.\nIf the bulk signal\ndeparts from the\nin vitro prediction,\ncan it be ascribed to\ncell composition?"
)
xq <- c(0.20, 0.50, 0.80)
for (i in seq_along(q)) {
  grid.draw(box_grob(xq[i], 0.41, 0.22, 0.13,
                     label = q[i], fill = "#FDEBD0", fontsize = 8.5))
}
for (i in seq_along(xq)) {
  grid.draw(arrow_grob(xq[i], 0.345, xq[i], 0.275, lwd = 0.7))
}

# ----- Stage 4 — principal finding -----
grid.text("STAGE 4 \u2014 Principal finding",
          x = 0.07, y = 0.255, hjust = 0,
          gp = gpar(fontsize = 10, fontface = "bold", col = "steelblue4"))

# Use plotmath so log2 / I^2 / ≈ render as true sub-/super-scripts and the
# arrow → as a real glyph. paste()/atop() are needed for multi-line math.
finding_label <- bquote(atop(
  atop(
    atop(
      "Tissue-bulk " * italic("IDI2-AS1") * " expression is invariant in every cohort (pooled " * log[2] * "FC " %~~% " 0; " * italic(I)^2 * " = 0.26 %).",
      "Sample-level " * italic("IDI2-AS1") * " " * symbol("\xab") * " " * italic("IL5") * " covariation is POSITIVE \u2014 opposite to the in vitro repressive direction."
    ),
    atop(
      "Causal mediation in the asthma cohort attributes " %~~% " 90 % of the covariation to the eosinophil compartment",
      "(ACME = +0.109, " * italic(p) * " = 0.002), with a within-tissue direct effect indistinguishable from zero (ADE = +0.012, " * italic(p) * " = 0.76)."
    )
  ),
  atop(
    "  ",
    atop(
      "\u21d2  The published in vitro " * italic("IDI2-AS1") %->% italic("IL5") * " axis is MASKED, not contradicted, by bulk RNA-seq.",
      "\u21d2  Single-cell follow-up is the natural next step."
    )
  )
))

grid.draw(box_grob(0.5, 0.135, 0.86, 0.17,
                   label = finding_label,
                   fill = "#E8F8F5", col = "darkgreen", lwd = 1.5,
                   fontsize = 9, fontface = "plain"))

popViewport()
dev.off()

cat("Wrote", out_pdf, "\n")

File: code/python/11_fig1_study_design.py
"""Fig 1 — Study design schematic (matplotlib version).

A4-landscape vector PDF (11.69 x 8.27 in). Single-column variant also produced
for Frontiers (single-column fig ~85 mm wide).

Layout: 4 stages stacked top-to-bottom
  Stage 1: 5 datasets (4 diseases, n = 856)
  Stage 2: uniform pipeline (DESeq2/limma -> meta -> corr -> mediation)
  Stage 3: three analytical questions
  Stage 4: principal finding (composition mediation ~= 90 %)
"""

from pathlib import Path

import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
from matplotlib.patches import FancyArrowPatch

plt.rcParams["font.family"] = "Arial"
plt.rcParams["pdf.fonttype"] = 42  # TrueType-embedded fonts
plt.rcParams["ps.fonttype"] = 42

OUT_PDF = Path(__file__).resolve().parents[2] / "results" / "figures" / "Fig1_study_design.pdf"
OUT_PDF.parent.mkdir(parents=True, exist_ok=True)


def draw_box(ax, x, y, w, h, label, *, fc="white", ec="black", lw=1.0,
             fontsize=8, fontweight="normal"):
    """Draw a centered rounded box with a label (x, y = center)."""
    rect = mpatches.FancyBboxPatch(
        (x - w / 2, y - h / 2), w, h,
        boxstyle="round,pad=0.002,rounding_size=0.008",
        linewidth=lw, edgecolor=ec, facecolor=fc,
    )
    ax.add_patch(rect)
    ax.text(x, y, label, ha="center", va="center",
            fontsize=fontsize, fontweight=fontweight,
            linespacing=1.15)


def draw_arrow(ax, x0, y0, x1, y1, lw=0.9, color="black"):
    arr = FancyArrowPatch(
        (x0, y0), (x1, y1),
        arrowstyle="-|>", mutation_scale=9,
        linewidth=lw, color=color,
        shrinkA=0, shrinkB=0,
    )
    ax.add_patch(arr)


def build_figure():
    fig = plt.figure(figsize=(11.69, 8.27))  # A4 landscape
    ax = fig.add_axes([0.02, 0.03, 0.96, 0.94])
    ax.set_xlim(0, 1)
    ax.set_ylim(0, 1)
    ax.set_axis_off()

    # -------- Title --------
    ax.text(0.5, 0.975,
            "Cross-disease re-analysis of the IDI2-AS1 / IL5 axis in IL5-driven allergic disease",
            ha="center", va="center",
            fontsize=13.5, fontweight="bold")
    ax.text(0.5, 0.942,
            "Five public bulk RNA-seq cohorts, 856 samples, uniform pipeline",
            ha="center", va="center",
            fontsize=10, style="italic", color="#444444")

    # -------- Stage 1: datasets --------
    ax.text(0.07, 0.89, "STAGE 1 — Datasets",
            ha="left", va="center",
            fontsize=10.5, fontweight="bold", color="#2874A6")

    ds = [
        ("Asthma\nGSE152004\nn = 695\nNasal epithelium",     "#D6EAF8"),
        ("Atopic dermatitis\nGSE121212\nn = 65\nSkin biopsy", "#FADBD8"),
        ("EoE\nGSE246323\nn = 10\nEsophagus",                 "#FCF3CF"),
        ("EoE\nGSE58640\nn = 16\nEsophagus",                  "#FCF3CF"),
        ("CRSwNP\nGSE136825\nn = 70\nNasal polyp",            "#D5F5E3"),
    ]
    xs = [0.12 + 0.19 * i for i in range(5)]
    y_stage1 = 0.81
    for (label, fc), x in zip(ds, xs):
        draw_box(ax, x, y_stage1, 0.165, 0.105, label, fc=fc, fontsize=8)

    # aggregator bar Stage 1 -> Stage 2
    bar_y = 0.745
    ax.plot([xs[0], xs[-1]], [bar_y, bar_y], color="#666666", lw=0.8)
    for x in xs:
        ax.plot([x, x], [y_stage1 - 0.052, bar_y], color="#666666", lw=0.8)
    draw_arrow(ax, 0.5, bar_y, 0.5, 0.712, lw=1.0, color="#444444")

    # -------- Stage 2: pipeline --------
    ax.text(0.07, 0.69, "STAGE 2 — Uniform pipeline",
            ha="left", va="center",
            fontsize=10.5, fontweight="bold", color="#2874A6")

    pipe = [
        "DESeq2 + apeglm\n(raw counts)\n/ limma-trend\n(FPKM)",
        "Random-effects\nmeta-analysis\n(metafor REML)",
        "Sample-level\nSpearman\nIDI2-AS1 ↔ IL5",
        "Cell-type-adjusted\npartial correlation\n(eosinophil + Th2\nsignatures)",
        "Causal mediation\n(asthma cohort,\n1,000 bootstraps)",
    ]
    xp = [0.12 + 0.19 * i for i in range(5)]
    y_stage2 = 0.61
    box_w = 0.175
    for label, x in zip(pipe, xp):
        draw_box(ax, x, y_stage2, box_w, 0.105,
                 label, fc="#EAF2F8", fontsize=7.5)
    for i in range(len(xp) - 1):
        draw_arrow(ax, xp[i] + box_w / 2, y_stage2,
                   xp[i + 1] - box_w / 2, y_stage2, lw=0.9, color="#444444")

    # -------- Stage 3: three questions --------
    ax.text(0.07, 0.49, "STAGE 3 — Three in vivo questions",
            ha="left", va="center",
            fontsize=10.5, fontweight="bold", color="#2874A6")

    q = [
        "Q1.\nIs IDI2-AS1 itself\ndifferentially expressed\nin disease tissue?",
        "Q2.\nDo IDI2-AS1 and IL5\ncovary at the sample level\nin the predicted\n(negative) direction?",
        "Q3.\nIf the bulk signal\ndeparts from the\nin vitro prediction,\ncan it be ascribed to\ncell composition?",
    ]
    xq = [0.22, 0.50, 0.78]
    y_stage3 = 0.41
    for label, x in zip(q, xq):
        draw_box(ax, x, y_stage3, 0.22, 0.125,
                 label, fc="#FDEBD0", fontsize=8.5)
    for x in xq:
        draw_arrow(ax, x, y_stage3 - 0.068, x, 0.285, lw=0.9, color="#444444")

    # -------- Stage 4: principal finding --------
    ax.text(0.07, 0.265, "STAGE 4 — Principal finding",
            ha="left", va="center",
            fontsize=10.5, fontweight="bold", color="#2874A6")

    finding = (
        "Tissue-bulk IDI2-AS1 expression is invariant in every cohort "
        "(pooled log$_2$FC ≈ 0; $I^2$ = 0.26 %).\n"
        "Sample-level IDI2-AS1 ↔ IL5 covariation is POSITIVE — opposite to the in vitro repressive direction.\n"
        "Causal mediation in the asthma cohort attributes ≈ 90 % of the covariation to the eosinophil compartment\n"
        "(ACME = +0.109, $p$ = 0.002), with a within-tissue direct effect indistinguishable from zero (ADE = +0.012, $p$ = 0.76).\n"
        "\n"
        "→ The published in vitro IDI2-AS1 → IL5 axis is MASKED, not contradicted, by bulk RNA-seq.\n"
        "→ Single-cell follow-up is the natural next step."
    )
    finding_box = mpatches.FancyBboxPatch(
        (0.07, 0.05), 0.86, 0.17,
        boxstyle="round,pad=0.004,rounding_size=0.012",
        linewidth=1.6, edgecolor="#1E6B3B", facecolor="#E8F8F5",
    )
    ax.add_patch(finding_box)
    ax.text(0.5, 0.135, finding, ha="center", va="center",
            fontsize=9, linespacing=1.4)

    return fig


def main():
    fig = build_figure()
    fig.savefig(OUT_PDF, format="pdf", bbox_inches="tight")
    # also a 300 dpi PNG for preview
    png = OUT_PDF.with_suffix(".png")
    fig.savefig(png, format="png", dpi=300, bbox_inches="tight")
    plt.close(fig)
    print(f"Wrote {OUT_PDF}")
    print(f"Wrote {png}")


if __name__ == "__main__":
    main()
